### DISCUSSION TITLE
Add support for company and customer

### DISCUSSION THREAD

**Discussion by blavka** [Document-Archiver/com.sophisticatedapps.archiving.document-archiver#16] [Ideas]
Hello, you are considering adding support for companies and it would be possible to have a scheme:
```
└── ~/Documents/DocumentArchiver
    └──#Customer#
       └── #Filetype#
           └── #Year#
               └── #Date#--#Description#__#Tags#.#Suffix#
```

Or another design, allow you to define the structure.

**Comment by stephansann** [comment]
Hi @blavka 

In deed multi tenancy support is something that was already on my mind. Thanks for bringing it up.

I opened an enhancement issue for this:
#17 

Best regards
Stephan

**Comment by stephansann** [comment]
BTW you can already set the archiving folder in the settings. So for as of now you have a way to separate documents for different tenants. Not comfortable, but a workaround.

**Reply by blavka** [reply]
Thanks, I found this settings, but as you say, it is't comfortable.

**Comment by stephansann** [comment]
Hi @blavka 

The newest SNAPSHOT-build contains support for tenants:

https://repository.sophisticatedapps.com/snapshots/com/sophisticatedapps/archiving/document-archiver/2.2.0-SNAPSHOT/document-archiver-2.2.0-20210520.095023-2-linux.zip
https://repository.sophisticatedapps.com/snapshots/com/sophisticatedapps/archiving/document-archiver/2.2.0-SNAPSHOT/document-archiver-2.2.0-20210520.095023-2-mac.zip
https://repository.sophisticatedapps.com/snapshots/com/sophisticatedapps/archiving/document-archiver/2.2.0-SNAPSHOT/document-archiver-2.2.0-20210520.095023-2-win.zip

(Select your platform -> Startup instructions on https://github.com/Document-Archiver/com.sophisticatedapps.archiving.document-archiver)

You just need to create 1 to n Subfolders with the naming "@TenantName" within the default Archiving folder:

```
└── ~/Documents/DocumentArchiver
    └──@TenantOne
       └── #Filetype#
           └── #Year#
               └── #Date#--#Description#__#Tags#.#Suffix#
```

and with the next start of Document Archiver you can select the tenant from the menu:

<img width="323" alt="Screen Shot 2021-05-20 at 11 06 03" src="https://user-images.githubusercontent.com/9678681/118961578-c05b2900-b964-11eb-997d-b1763177f0c6.png">

Later I will add a way to create (and delete) tenants from within Document Archiver. Anyhow like this you can already start working with more comfort.

Best regards
Stephan

**Reply by blavka** [reply]
Nice I try. 
THX

**Comment by stephansann** [comment]
Hi @blavka 

The newest SNAPSHOT build now also includes tenant management:
https://repository.sophisticatedapps.com/snapshots/com/sophisticatedapps/archiving/document-archiver/2.2.0-SNAPSHOT

Let me know if it works for you and what you think.

Best regards
Stephan
